src/modules/NetworkController/private/Copy-CertificatesToFabric.ps1

```diff
@@ -0,0 +1,142 @@
+        [System.Security.SecureString]$CertPassword,
+
+        [Parameter(Mandatory = $true)]
+        [System.Object]$FabricDetails,
+
+        [Parameter(Mandatory = $false)]
+        [Switch]$RotateNodeCertificates,
+
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty
+    )
+
+    begin {
+        $certificateCache = @()
+        $certificateConfig = @{
+            NetworkController = @{}
+        }
+
+        $nodesToRotateRestCert = @()
+        $nodesToRotateRestCert += $FabricDetails.NetworkController
+        $nodesToRotateRestCert += $FabricDetails.SoftwareLoadBalancer
+        $nodesToRotateRestCert += $FabricDetails.Server
+
+        [System.String]$certDir = "$(Get-WorkingDirectory)\RotateCert"
+
+        "Scanning certificates within {0}" -f $CertPath.FullName | Trace-Output
+        $pfxFiles = Get-ChildItem -Path $CertPath.FullName -Filter '*.pfx'
+        if ($null -eq $pfxFiles) {
+            throw New-Object System.NullReferenceException("Unable to locate .pfx files under the specified CertPath location")
+        }
+
+        "Retrieving Rest Certificate" | Trace-Output
+        $currentRestCertificate = Get-SdnNetworkControllerRestCertificate
+
+        "Creating {0} directory for certificate staging on {1}" -f $certDir, ($nodesToRotateRestCert -join ', ') | Trace-Output
+        Invoke-PSRemoteCommand -ComputerName $nodesToRotateRestCert -Credential $Credential -ScriptBlock {
+            $null = New-Item -Path $using:certDir -ItemType Directory -Force
+        } -ErrorAction Stop
+    }
+
+    process {
+        foreach ($pfxFile in $pfxFiles) {
+            "Retrieving PfxData for {0}" -f $pfxFile.FullName | Trace-Output
+            $pfxData = Get-PfxData -FilePath $pfxFile.FullName -Password $CertPassword -ErrorAction Stop
+
+            $object = [PSCustomObject]@{
+                FileInfo = $pfxFile
+                PfxData = $pfxData
+            }
+
+            $certificateCache += $object
+        }
+
+        # enumerate the current certificates within the cache to isolate the rest certificate
+        foreach ($cert in $certificateCache) {
+            if ($cert.pfxdata.EndEntityCertificates.Subject -ieq $currentRestCertificate.Subject) {
+                "Matched {0} [Subject: {1}; Thumbprint: {2}] to NC Rest Certificate" -f `
+                $cert.pfxFile.FileInfo.FullName,  $cert.pfxData.EndEntityCertificates.Subject, $cert.pfxData.EndEntityCertificates.Thumbprint | Trace-Output
+
+                $restCertificate = $cert
+                break
+            }
+        }
+
+        # install the rest certificate to Network Controllers, MUXes and Servers
+        # if the certificate is self signed, it will automatically install to the root store as part of the process
+        if ($restCertificate) {
```

Comment:
Will this install cert with private key into MY store? If that is the case, is it proper to do so? Should we also export rest cert into .cer format and install it to MUX, Servers root instead of MY

Reply:
Let me research a bit and get back to you on this.